src/plugins/teleport-plugin-react-css-modules/index.ts

```diff
@@ -65,16 +65,13 @@ export const createPlugin: ComponentPluginFactory<ReactCSSModulesConfig> = (conf
         if (Object.keys(dynamicStyles).length) {
           const rootStyles = cleanupNestedStyles(dynamicStyles)
 
-          const inlineStyles = transformDynamicStyles(
-            rootStyles,
-            (styleValue) =>
-              new ParsedASTNode(
-                t.arrowFunctionExpression(
-                  [t.identifier('props')],
-                  t.memberExpression(t.identifier('props'), t.identifier(styleValue.content.id))
-                )
-              )
-          )
+          const inlineStyles = transformDynamicStyles(rootStyles, (styleValue) => {
+            const expression =
```

Comment:
we should do a switch based function here, in case it's not `state` and not `props`